src/Raven.Server/NotificationCenter/RequestLatency.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Session;
+using Raven.Server.NotificationCenter.Notifications;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Sparrow.Json;
+
+namespace Raven.Server.NotificationCenter
+{
+    public class RequestLatency
+    {
+        private static readonly string RequestLatenciesId = $"{NotificationType.PerformanceHint}/{PerformanceHintType.RequestLatency}";
+
+        private readonly NotificationCenter _notificationCenter;
+        private readonly NotificationsStorage _notificationsStorage;
+        
+        public RequestLatency(NotificationCenter notificationCenter, NotificationsStorage notificationsStorage)
+        {
+            _notificationCenter = notificationCenter;
+            _notificationsStorage = notificationsStorage;
+        }
+
+        public void AddHint(string queryString, long duration, string action)
+        {
+            var requestLatencyPerformanceHint = GetOrCreatePerformanceLatencies(out var details);
```

Comment:
This make me nervous on the ground of multi threaded behavior.
How does this behave in this regard?